uwdata/lyra
# Undo/Redo

**Description of the problem this pull request fixes**
Track history of definitions of visualization components to enable undo/redo. Application state (inspectors, walkthroughs, hints, etc.) are not part of the history. 

**Changes proposed in this pull request:**
- Reshape store to group together all "undoable" states, and use [redux-undo](https://github.com/omnidan/redux-undo) to add history tracking.

**This PR includes:**

**Steps to functionally test this:**
- Follow #331 and use undo/redo buttons in the toolbar to test undo/redo functionality. In particular,
  - The application UI (sidebars) should update and the Vega view should also update.
  - Binding a data field to a dropzone/channel should only register as a single element in the history state.
  - Continuous changes should be throttled/debounced and only register as a single historical state (e.g., dragging or resizing a mark, dragging a color picker, changing a number field input for axes).
